redis.conf

```diff
@@ -70,6 +70,29 @@ tcp-backlog 511
 # unixsocket /tmp/redis.sock
 # unixsocketperm 700
 
+# Enable forcing of TCP connections over SSL. 
+# If enabled, you must specify either a certificate root store file or directory (not both) in 
+# PEM format (or a set of PEM files if a directory). You must also specify the certificate for the
+# server, the private key file, and the Diffie-Hellman parameters file. These may be in separate files
+# or in the same file. All files must be in PEM format (as required by openSSL). 
+# If you wish, you can customize the cipher list to use when accepting SSL connections. By default, the list
+# By default, we use all ciphers with the exception of: ones with anonymous Diffie-Hellman, low-strength ciphers, 
+# export ciphers, md5 hashing. Ordered from strongest to weakest.
+# Note that as ciphers become broken, it will be necessary to change the available cipher list to remain secure. 
+#
+# If you are using slave processes, and wish to validate the common name of the redis server, you can
+# specify the parameter ssl_cert_common_name with the name you wish to validate against.
+#
+# ssl true
+# ssl_ca_root_dir < Path to your openSSL root certificates directory >
+# ssl_ca_root_file < Path to your openSSL root certificate file (You must specify either a root CA file, or a directory, but not both) >
+# ssl_cert_file <Path to your server's SSL cert>
+# ssl_pk_file <Path to your server's private key file (may be the same as above)>
+# ssl_dhk_file <Path to your Diffie-Hellman data file (may be the same as above)>
+# ssl_cert_pass <optional - password for server cert file>
+# ssl_cipher_list <optional - cipher list to use for the server. Defaults to ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH>
+# ssl_cert_common_name <optional - common name of server cert to match>
+
```

Comment:
@bkerley I looked at the online documentation, and the methods they have available are:
 * SSLv2, SSLv3, TLS1, TLS1.1, TLS1.2
 * SSLv3 only
 * TLS1 only
 * TLS1.1 only

Now, the best I've seen (to allow the most flexibility) would be to specify the first one, which allows everything, then allow the configuration file to turn off modes that you don't want. There are options flags for SSL_NO_OP_<xxx> where <xxx> can specify the individual protocols.